src/System.Net.Security/src/System/Net/Unix/SSPISecureChannelType.cs

```diff
@@ -27,7 +27,15 @@ public void VerifyPackageInfo()
         public SafeFreeCredentials AcquireCredentialsHandle(X509Certificate certificate,
             SslProtocols protocols, EncryptionPolicy policy, bool isServer)
         {
-            return new SafeFreeCredentials(certificate, protocols, policy);
+            SafeFreeCredentials retVal = new SafeFreeCredentials(certificate, protocols, policy);
+            if ((null != retVal) && !retVal.IsInvalid)
+            {
+                // Caller does a ref count decrement
```

Comment:
But only if !IsInvalid?  I'd suggest not trying to optimize for the IsInvalid case and just always doing the AddRef/ReleaseRef, regardless of whether the value is valid or not (as long as it's not null).